src/helpers/processContractStorageData.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  AnyObject,
+  ContractStorageProcessedItem,
+  SortDirection,
+} from "@/types/types";
+
+export const processContractStorageData = <T extends AnyObject>({
+  data,
+  sortById,
+  sortByDir,
+  filters,
+}: {
+  data: T[];
+  sortById: string | undefined;
+  sortByDir: SortDirection;
+  filters: { [key: string]: string[] };
+}): ContractStorageProcessedItem<T>[] => {
+  let sortedData = [...data];
+
+  // Sort
+  if (sortById) {
+    if (["asc", "desc"].includes(sortByDir)) {
+      // Asc
+      sortedData = sortedData.sort((a: any, b: any) =>
+        a[sortById] > b[sortById] ? 1 : -1,
+      );
+
+      // Desc
+      if (sortByDir === "desc") {
+        sortedData = sortedData.reverse();
+      }
```

Comment:
for descending, instead of doing two operations by sorting via ascending then descending via `.reverse()`, can we just do it in one go? also should we include a case when there might be an equality between the value?

Reply:
That's a good point. :+1: I'll update to use just one operation for both cases.

I don't think we need to do anything special with the equal cases because we don't have a secondary param to sort by if they are equal. Right now, it doesn't matter which item comes first.